### ISSUE TITLE
Sprint 5th September - 16th september

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#11579]
# Global Sprint Planning

## 3 things that might take us down

Last week
1. Enforcing 2FA on github/gsuite/DO/NPM/Ruby gems. @timgl to do this. Mostly done, one or two left
2. Cap # of recordings/day or events/day. @rcmarron to write something up. Done

This week
1. Rate limiting API requests - @rcmarron to figure this out, just target clickhouse

## Retro: Status of Outcomes from Previous Sprint

1. Call the persons/groups on events done (from the ingestion side). Documentation/async migration/0.8% of events.  @tiina303 (documentation is live, reduced the percentage of events even further)
Why? To speed up queries and scale
4. Unblocking large historical exports, fix the current issues, async server is struggling, benchmarking. @macobo
Why? Enabling a large customer (got further than expected - fixed a bunch of bugs early and ended up doing a full rewrite of historical exports)
5. We are at 10% of events on EKS currently. Plan is to be at 100% by end of this sprint with as few 500s as possible. @hazzadous & @ellie
why? Dogfood helm chart
6. Improve onboarding experience for helm chart dev and docs @ellie
Why? Improve developer experience + will help contributions
7. Finalize and tune monitoring as code framework for helm chart @guidoiaquinti
Why? Baking in best practices for helm chart
8. Fix 40 bugs from https://github.com/PostHog/posthog/issues/11344. 4 engineers x 10 days. Status: 11 fixed, 9 ish coming. Want to continue next sprint
Why? Make sure we have a great experience + increase retention
9. Recordings experience enhancement around quality reporting + console log. @rcmarron. Status: Good progress, filtering will be next sprint
why? Make our session recording experience comprehensive/competitive
10. Complete person on events rollout @EDsCODE. Status: Implementation is done, but haven't fully rolled out.
why? To unlock scale & quicker queries
11. Feature Flags Client and UX update @neilkakkar -> Li is the MVP for getting most UX stuff done. I did none of the backend stuff, person-on-events had higher prio
Why? To make our feature flag competitive

## Retro: What can we do better next sprint?

1. We planned 40 engineering days without realising support hero. Maybe we should mention this before breakout
2. 
## Support hero this sprint

Week 1:  Yakko
Week 2: Guido

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. Add session recording billing @kappa90 
why: Because we want to charge for session recordings
2. Continue bugbashing. @mariusandra to provide a list on Friday. Timebox to this sprint https://github.com/orgs/PostHog/projects/58/
why: Make sure we have a great experience + increase retention
3. $create_alias deprecation @yakkomajuri 
why: to roll out persons on events
4. PostHog EU (ClickHouse / Zookeeper production ready, runbooks, + other services if possible) @ellie
why? to increase usage from customers that need EU cloud
5. Harry: investigate latency issues and random error rate decrease @hazzadous 
9. Session recording event log filtering in playlist  @alexkim205 
why? Make our session recording experience comprehensive/competitive
10. Complete person on events rollout @EDsCODE.
why? To unlock scale & quicker queries
12. Feature Flags: posthog-js and react-native bootstrapping, server side override, GeoIP overrides for everyone, and UX issues: audit history, feature flags on person @neilkakkar
Why? To make our feature flag competitive

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Bugs

-

### Low priority / side quests

-

```

**Comment by kappa90** [comment]
Team Growth

## Retro
- Ran tutorial experiment, but failed due to bugs
    - https://github.com/PostHog/posthog/pull/10956 
- Make it more obvious that self-hosting is more expensive than cloud
    - https://github.com/PostHog/posthog.com/pull/4083
- More heavily discourage using posthog as domain in deployment
    - https://github.com/PostHog/posthog.com/pull/4081
    - https://github.com/PostHog/posthog/issues/11533 -> probably not happening
- Add a “Where did you hear about us” field in the Signup form (needs to open PR for it, but ready)
- Checklist during activation to tell users suggested actions that can help them get the most out of PostHog
    - Opened an RFC https://github.com/PostHog/product-internal/pull/338
    - Will deploy a checklist by Friday, and in the next sprint we'll add marketing resources to it
    - Tutorial will be reimplemented as part of the checklist

## Hang over items from previous sprint
- Finalise adding card request page during ingestion/onboarding and update billing plans copy
- Need to push a change to the [hosting cost](https://posthog.com/docs/self-host/deploy/hosting-costs)

## Planning
- Finalise adding card request page during ingestion/onboarding and update billing plans copy
- Research usage based billing systems to replace Stripe/custom code

### High priority
- Add session recording billing together with @rcmarron 

### Low priority / side quests
- Implement dashboard templates

**Comment by mariusandra** [comment]
Team App East Dragons

## Retro

TODO

## Hang over items from previous sprint

- So far: Added 15 bugs, fixed 11.
- Will fix around 20 bugs by the end of the sprint. So net negative hopefully, but shy of the 40 bug goal.

Fixed several small bugs, and many big dragons:
- Annotations suck no more
- Persons improvements: search working, no missing, modal not horiffic
- CSV exports for all people
- Date filter lemonization
- We will not load 100MB of event/property definitions anymore

## Planning

Continue the bugbash, create a prioritised backlog that we keep up to date. https://github.com/orgs/PostHog/projects/58/
Why? Make sure we have a great experience + increase retention

- Prioritise the outstanding bugs
- Fix more bugs.
- Make sure there are less outstanding bugs at the end of the sprint compared to before

**Comment by yakkomajuri** [comment]
# Team Ingestion 

## Goals

1. Stay afloat @yakkomajuri 
Tasks to attempt to cover: outage follow-ups, $create_alias deprecation, graphile-worker robustness

Note: Karl is out, Tiina is out for at least a day + just had a pretty bad injury, Yakko is support hero

**Comment by guidoiaquinti** [comment]
# Team Infrastructure

## Retro

1. Ellie: we should take less on. Lots of unknown infra problems. Business as usual is more than expected.
2. Guido: we have a focus problem. From customer support requests, to internal eng requests, to quarterly tasks, to ...
3. Harry: N/A

## Hang over items from previous sprint

> We are at 10% of events on EKS currently. Plan is to be at 100% by end of this sprint with as few 500s as possible. @hazzadous & @ellie. Why? Dogfood helm chart

Still a priority but we've hit several roadbumps (latest are latency increase and error rate not directly related to this rollout)

## Planning

1. Ellie: PostHog EU (ClickHouse / Zookeeper production ready, runbooks, + other services if possible)
2. Guido: help with PostHog EU + business as usual + support hero
3. Harry: investigate latency issues and random error rate decrease

**Comment by EDsCODE** [comment]
Team App West

## Retro

TODO

## Planning

**Recordings experience enhancement** @rcmarron @alexkim205 
- Charging for recordings + upselling
- Recordings setup granularity
- New Filtering 

**Person on events rollout** @EDsCODE @neilkakkar 
- Validation finally
- Improve monitoring

**Feature Flags** @neilkakkar @liyiy 
- Client library updates: Enable bootstrap for posthog-js and react-native
- Passing person and group properties to decide
- Decide server prop overrides & GeoIP for everyone
- Evaluation reasons for flags
- UX issues Complete what’s remaining
